api/app/controllers/spree/api/v2/storefront/account/addresses_controller.rb

```diff
@@ -16,6 +16,14 @@ def update
               render_result(result)
             end
 
+            def destroy
+              if resource.destroy
```

Comment:
```suggestion
              spree_authorize! :destroy, resource

              if resource.destroy
```

let's also add this `authorize` check for update as well